test/regressionmodel.jl

```diff
@@ -6,13 +6,31 @@ using StatsAPI: RegressionModel, crossmodelmatrix
 struct MyRegressionModel <: RegressionModel
 end
 
+struct MyWeightedRegressionModel <: RegressionModel
+    wts::AbstractVector
+end
+
 StatsAPI.modelmatrix(::MyRegressionModel) = [1 2; 3 4]
 
+function StatsAPI.modelmatrix(r::MyWeightedRegressionModel; weighted::Bool=false)
+    X = [1 2; 3 4]
+    weighted ? sqrt.(r.wts).*X : X
+end
+
+w = [0.3, 0.2]
```

Comment:
Given that the methods hardcode the matrix, probably not worth having a separate type which doesn't hardcode weights:
```suggestion
function StatsAPI.modelmatrix(r::MyRegressionModel; weighted::Bool=false)
    X = [1 2; 3 4]
    w = [1.5, 2.0, 0.3, 3.5]
    weighted ? sqrt.(w).*X : X
end
```
... and simplify tests below.